client/packages/inventory/src/Stocktake/ListView/CreateStocktakeModal.tsx

```diff
@@ -0,0 +1,268 @@
+import React, { FC, useState } from 'react';
+import {
+  BasicSpinner,
+  Checkbox,
+  DateTimePickerInput,
+  DialogButton,
+  InputWithLabelRow,
+  Typography,
+} from '@common/components';
+import { useTranslation } from '@common/intl';
+import { useDialog } from '@common/hooks';
+import {
+  useStockList,
+  useLocationList,
+  useMasterLists,
+  LocationSearchInput,
+  LocationRowFragment,
+  MasterListSearchInput,
+  MasterListRowFragment,
+} from '@openmsupply-client/system';
+import { Box, useAuthContext } from '@openmsupply-client/common';
+import { useCreateStocktake } from './createStocktake';
+
+const LABEL_FLEX = '0 0 150px';
+interface CreateStocktakeArgs {
+  masterListId?: string;
+  locationId?: string;
+  itemsHaveStock?: boolean;
+  expiresBefore?: Date | null;
+  isInitialStocktake: boolean;
+  comment: string | undefined;
+}
+
+const DEFAULT_ARGS: CreateStocktakeArgs = {
+  masterListId: '',
+  locationId: '',
+  itemsHaveStock: false,
+  expiresBefore: null,
+  isInitialStocktake: false,
+  comment: undefined,
+};
```

Comment:
Would consolidate this with CreateStocktakeArgs and DEFAULT_ARGS with `CreateStocktakeParams`